core/lib/zksync_core/src/basic_witness_input_producer/vm_interactions.rs

```diff
@@ -0,0 +1,83 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use crate::state_keeper::io::common::load_l1_batch_params;
+
+use multivm::vm_latest::HistoryEnabled;
+use multivm::VmInstance;
+use tokio::runtime::Handle;
+use zksync_dal::StorageProcessor;
+use zksync_state::{PostgresStorage, ReadStorage, StorageView};
+use zksync_types::{L1BatchNumber, L2ChainId, Transaction};
+
+pub(super) fn create_vm(
+    rt_handle: Handle,
+    l1_batch_number: L1BatchNumber,
+    mut connection: StorageProcessor<'_>,
+    validation_computational_gas_limit: u32,
+    l2_chain_id: L2ChainId,
+) -> (
+    VmInstance<PostgresStorage, HistoryEnabled>,
+    Rc<RefCell<StorageView<PostgresStorage>>>,
+) {
+    let prev_l1_batch_number = l1_batch_number - 1;
+    let (_, miniblock_number) = rt_handle
+        .block_on(
+            connection
+                .blocks_dal()
+                .get_miniblock_range_of_l1_batch(prev_l1_batch_number),
+        )
+        .unwrap()
+        .unwrap_or_else(|| {
```

Comment:
Nit: `process_job_impl` returns `anyhow::Result`, so all the unwraps here can be replaced with corresponding errors.